include/wx/vector.h

```diff
@@ -219,6 +219,8 @@ class wxVector
             { return m_ptr == it.m_ptr; }
         bool operator !=(const reverse_iterator& it) const
             { return m_ptr != it.m_ptr; }
+        bool operator<(const reverse_iterator& it) const
```

Comment:
Sorry but why exactly is this necessary? Do we use sets/maps of iterators somewhere?

Reply:
In debug mode under Visual Studio, the std vector implementation checks that the iterators are oriented properly. It uses the "<" operator for this. So iterators need to support it. Without that, the code in the test won't even compile.

For example. in C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\xutility, there is a function called _Debug_range2, which tests an iterator pair for a valid range. Without the operator<, it won't compile. It works fine in release mode, where these checks don't kick in.

Reply:
When I commented out that part of the vector test for random iterators, the rest of the code built, which makes me wonder if the test is testing something the rest of the system doesn't even care about. That could be an alternate fix, if that capability isn't even needed.

Reply:
OK, thanks, I see. Weird that this doesn't happen neither with previous MSVS versions (tested on AppVeyor) nor with VS 14 or 14.1 which I use, but only with VS 12, but it still does need to be fixed. We really need to define all the operators, however, and not just `<`, if we define any of them, otherwise it's just too weird. I'll do it soon, thanks for reporting this.